### PR TITLE
[TT-10357] Add fallback_to_default field to version def

### PR DESCRIPTION
**Pull request by furkansenharputlu**
parent ticket: https://tyktech.atlassian.net/browse/TT-9901
subticket: https://tyktech.atlassian.net/browse/TT-10357

### CODE REVIEW

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5678/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6645542291)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5678/merge
Commit: a2a3f4ea90727b597ffe52ef47608be7077968d1 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6647419660)

File: gateway/mw_version_check.go [hunk 1]
```diff
@@ -68,22 +68,45 @@ func (v *VersionCheck) ProcessRequest(w http.ResponseWriter, r *http.Request, _
...
 
 		subVersionID := v.Spec.VersionDefinition.Versions[targetVersion]
 		handler, _, found := v.Gw.findInternalHttpHandlerByNameOrID(subVersionID)
 		if !found {
-			return errors.New(string(VersionDoesNotExist)), http.StatusNotFound
+			if !v.Spec.VersionDefinition.FallbackToDefault {
+				return errors.New(string(VersionDoesNotExist)), http.StatusNotFound
+			}
+
+			if isBase(v.Spec.VersionDefinition.Default) {
```
Comment (jeffy-mathew) [inline gateway/mw_version_check.go:87]: can we invert this condition to have the statements after this block so that we can avoid the use of goto?

Comment (furkansenharputlu) [reply gateway/mw_version_check.go:87]: It is to prevent lookup

Comment (furkansenharputlu) [reply gateway/mw_version_check.go:87]: Can you please write in more dtail?

**Review by jeffy-mathew — APPROVED** [review]

**Pull request merged**